File: agent_golang/model/serviceinfo.go
package model

type ServiceInfo struct {
	System            string `json:"system"`
	DeploymentID      string `json:"deploymentId"`
	Address           string `json:"address"`
	Port              int    `json:"port"`
	HealthEndpoint    string `json:"healthEndpoint"`
	IsBalancingEnabled bool  `json:"isBalancingEnabled"`
}

File: common/src/main/java/com/wandern/model/ServiceInfo.java
package com.wandern.model;

import com.wandern.clients.ServiceStatus;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record ServiceInfo(
    String deploymentId,
    String deploymentUnit,
    String system,
    String serviceUrl,
    String contextPath,
    int port,
    String ip,
    ServiceStatus status,
    String programmingLanguage,
    LocalDateTime registrationTime
) {}